lib/ocpp/v201/database_handler.cpp

```diff
@@ -585,5 +585,19 @@ void DatabaseHandler::insert_or_update_charging_profile(const int evse_id, const
     }
 }
 
+void DatabaseHandler::delete_charging_profile(const int profile_id) {
+    std::string sql = "DELETE FROM CHARGING_PROFILES WHERE ID = @profile_id;";
+    auto stmt = this->database->new_statement(sql);
+
+    stmt->bind_int("@profile_id", profile_id);
+    if (stmt->step() != SQLITE_DONE) {
+        EVLOG_error << "Could not delete from table: " << this->database->get_error_message();
+    }
+}
```

Comment:
The `id` in the table is set to the `chargingProfile.id` which is unique. See K01.FR.05, K01.FR.52 and K01.FR.53. I believe this is an acceptable way to delete.

Reply:
K01.FR.05 is a bit confusing - our [`add_profile()`](https://github.com/US-JOET/libocpp/blob/feature/k01-add-profile/lib/ocpp/v201/smart_charging.cpp#L200C1-L210C6) method lets us have a profile with the same id as long as its on either the EVSE or charging station. This is because the requirement definition has some ambiguity:

> The Charging Station SHALL replace the existing ChargingProfile with the one specified.

Since it would not make any sense to replace a station-wide profile with an EVSE-specific one in this scenario (or vice-versa), the only thing that seems logical is to separate EVSE-specific profiles from station wide profiles.

Reply:
@christopher-davis-afs not seeing how this is directly relevant to the card. Feels like an issue to track, and talk with the larger community, but don't see it blocking this.

Reply:
I agree, hence I marked it as a non-blocking comment and approved the PR :)